contrib/coredns/policy/policy.go

```diff
@@ -150,7 +174,47 @@ func (p *PolicyMiddleware) ServeDNS(ctx context.Context, w dns.ResponseWriter, r
 	}
 
 	if response.Permit {
```

Comment:
If you want to create a separate function for this stuff for unit testing purposes (as you did before), then I think it's OK, but you should call it something like `handlePermit`, and it should take the same params as `ServeDNS`, plus the policy response.